### ISSUE TITLE
ncat: invalid option -- 'z'

### ISSUE THREAD

**Issue by zoowar** [sandstorm-io/sandstorm#973]
I have successfully installed sandstorm on my system in the past. Today I decided to remove /opt/sandstorm and do it again. The installation failed to complete (see log below). The server is active and running.

Ncat 6.47
Fedora 22

````
<snip>
Config written to /opt/sandstorm/sandstorm.conf.
Finding latest build for dev channel...
Downloading: https://dl.sandstorm.io/sandstorm-117.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21.0M  100 21.0M    0     0   311k      0  0:01:09  0:01:09 --:--:--  310k
GPG signature is valid.
Created symlink from /etc/systemd/system/multi-user.target.wants/sandstorm.service to /etc/systemd/system/sandstorm.service.
Your server is coming online. Waiting up to 90 seconds...nc: invalid option -- 'z'
Ncat: Try `--help' or man(1) ncat for more information, usage options and help. QUITTING.
<snip>
.nc: invalid option -- 'z'
Ncat: Try `--help' or man(1) ncat for more information, usage options and help. QUITTING.
Your server never started listening.
*** INSTALLATION FAILED ***
Report bugs at: http://github.com/sandstorm-io/sandstorm
````

**Comment by paulproteus** [comment]
Hi zoowar!

Thanks hugely for this bug report. I'll add a test for Fedora to the
installer-tests/ system and fix this bug.

Out of curiosity, what does 'rpm -q nc' and 'rpm -q netcat' say? And what
version of Fedora is this?

Cheerio, Asheesh.

**Comment by zarvox** [comment]
I can reproduce this.

Looks like the installed `/usr/bin/nc` comes from `nmap-ncat`, for me on F22.

**Comment by paulproteus** [comment]
Sweet. I have that on Debian as /usr/bin/ncat and can see if it supports a
"zero i/o" (scanning) mode.

**Comment by zoowar** [comment]
@zarvox is correct: nmap-ncat-2:6.47-2.fc22.x86_64 : Nmap's Netcat replacement

**Comment by paulproteus** [comment]
Looks like I can build the same thing on top of:

/usr/bin/ncat --send-only localhost 22 < /dev/null

**Comment by zarvox** [comment]
Cool!  Sounds like you can version-detect what `nc` is and pick some `$NC_ARGS` to pass in at the callsites.

**Comment by zoowar** [comment]
For the record: The work around is to verify sandstorm is running with `systemctl status sandstorm.service` and generate an admin token with `sudo /opt/sandstorm/sandstorm admin-token`. Then complete the installation by visiting the admin settings url provide in the previous command.

**Comment by paulproteus** [comment]
Hi @zoowar or @zarvox ,

Could you confirm that https://raw.githubusercontent.com/sandstorm-io/sandstorm/support-different-netcats/install.sh passes a sanity-check on your systems?

It sure works great on mine, so I think I'm OK merging it as-is, but feedback is always nice.

Particularly:

* On a system with port 80 & 443 available, does it successfully tell you that it will give you HTTPS?

* On a system without port 80&443 available, does it successfully _not_ tell you that it will give you HTTPS?


Cheerio, Asheesh.

**Comment by zoowar** [comment]
I'm not sure what you're asking for. The only protocol indication is the link to the admin settings: https://<redacted>.sandcats.io/admin/settings/<redacted>. Sandstorm does listen on ports 80 and 443 when running. Hope this helps.

**Comment by paulproteus** [comment]
Thanks @zoowar . I think I'll mark this as closed with #981 !
